src/library/searchquery.cpp

```diff
@@ -166,6 +172,46 @@ QString TextFilterNode::toSql() const {
     return concatSqlClauses(searchClauses, "OR");
 }
 
+CrateFilterNode::CrateFilterNode(const QSqlDatabase& database,
+                                 const QString& argument)
+    : m_database(database),
+      m_argument(argument) {
+    getTrackIds();
+}
+
+void CrateFilterNode::getTrackIds() {
+    FieldEscaper escaper(m_database);
+    QString escapedArgument = escaper.escapeString(kSqlLikeMatchAll + m_argument + kSqlLikeMatchAll);
+
+    QSqlQuery query(m_database);
```

Comment:
@uklotzde I have added the following to `CrateFilterNode::init()`
`FieldEscaper escaper(m_pTrackCollection->database());
QString escapedArgument = escaper.escapeString(kSqlLikeMatchAll + m_argument + kSqlLikeMatchAll);
CrateTrackSelectResult crateTracks(m_pTrackCollection->crates().selectTracksSortedByCrateNameLike(escapedArgument));
while (crateTracks.next()) {
    m_trackIds.push_back(crateTracks.trackId());
}`

`m_trackIds` is a vector.

I get an error:
`Warning [Main]: QSqlQuery::value: not positioned on a valid record
Critical [Main]: Invalid database identifier value: QVariant(, )`
from the while loop.

Do you have any idea why?